stream_alert/rule_processor/handler.py

```diff
@@ -124,35 +138,71 @@ def run(self, event):
         # Check if debugging logging is on before json dumping alerts since
         # this can be time consuming if there are a lot of alerts
         if self._alerts and LOGGER.isEnabledFor(log_level_debug):
-            LOGGER.debug('Alerts:\n%s', json.dumps(self._alerts, indent=2))
+            LOGGER.debug('Alerts:\n%s', json.dumps(self.get_alerts(), indent=2))
 
         # Send any cached metrics to CloudWatch before returning
         self.metrics.send_metrics()
 
-        return self._failed_record_count == 0
+        return self._failed_record_count.value == 0
 
     def get_alerts(self):
         """Public method to return alerts from class. Useful for testing.
 
         Returns:
             [list] list of alerts as dictionaries
         """
-        return self._alerts
+        return self._alerts._getvalue()
 
-    def _process_alerts(self, payload):
-        """Process records for alerts and send them to the correct places
+    def _run_batches(self, payload, workers):
+        """Get all of the pre-parsed records from the payload and segment them
+        into chunks to be spread accross multiprocessing workers.
 
         Args:
             payload [StreamPayload]: StreamAlert payload object being processed
+            workers [list<mutliprocessing.Process>]: The complete list of
+                multiprocessing jobs to append this job to
+        """
+        # Get all of the pre-parsed records for this payload
+        payload_records = [record for record in payload.pre_parse()]
+
+        # Get the interval that we should segment the list of alerts on and
+        # arrange them into sublists of this size. Default to minimum of 1
```

Comment:
If you create a `multiprocessing.Pool`, I don't think you need to explicitly segment the list; it should handle it automatically.